media/linux/pds-sqlite3-queries/training_sheets.py

```diff
@@ -0,0 +1,166 @@
+
+from oauth2client import tools
+from apiclient.http import MediaFileUpload
+
+from openpyxl import Workbook
+from openpyxl.styles import Font, PatternFill, Alignment
+
+from pprint import pprint
+from pprint import pformat
+
+now = datetime.now()
+
+class MetaSheet:
+
+    def __init__(self, ws, trainingdata):
+        self.ws = ws
+        self.values = trainingdata
+        self.roster = list()
+
+    def create_sheet(self, title, want_everything):
+        ws = self.ws
+        title_font = Font(color='FFFF00')
+        title_fill = PatternFill(fgColor='0000FF', fill_type='solid')
+        title_align = Alignment(horizontal='center')
+
+        last_col = 'K'
+    
+        row = 1
+        ws.merge_cells(f'A{row}:{last_col}{row}')
+        cell = f'A{row}'
+        ws[cell] = f'Training: {title}'
+        ws[cell].fill = title_fill
+        ws[cell].font = title_font
+
+        row = row + 1
+        ws.merge_cells(f'A{row}:{last_col}{row}')
+        cell = f'A{row}'
+        ws[cell] = f'Last updated: {now}'
+        ws[cell].fill = title_fill
+        ws[cell].font = title_font
+
+        row = row + 1
+        ws.merge_cells(f'A{row}:{last_col}{row}')
+        cell = f'A{row}'
+        ws[cell] = ''
+        ws[cell].fill = title_fill
+        ws[cell].font = title_font
+
+        row = row + 1
+        columns = [ (f'A{row}', 'Start Date'             ,   30),
+                    (f'B{row}', 'End Date'               ,   30),
+                    (f'C{row}', 'Member Name'            ,   30),
+                    (f'D{row}', 'Email Address'          ,   30),
+                    (f'E{row}', 'Phone Number'           ,   50),
+                    (f'F{row}', 'Weekend?'               ,   50),
+                    (f'G{row}', 'Weekday?'               ,   50),
+                    (f'H{row}', 'Homebound?'             ,   50),
+                    (f'I{row}', 'Notes'                  ,   50),]
+        if want_everything:
+            columns.append((f'J{row}', 'Stage of Certification', 50))
+            columns.append((f'K{row}', 'Active Parishoner?'    , 50))
+
+        for cell,value,width in columns:
+            ws[cell] = value
+            ws[cell].fill = title_fill
+            ws[cell].font = title_font
+            ws[cell].alignment = title_align
+            ws.column_dimensions[cell[0]].width = width
+
+        row = row + 1
+        ws.freeze_panes = ws[f'A{row}']
+
+        return row, ws
+
+    def create_entry(self, ws, entry, row):
+        ws[f'A{row}'] = entry['start_date']
+        ws[f'B{row}'] = entry['end_date']
+        ws[f'C{row}'] = entry['name']
+        ws[f'D{row}'] = entry['email']
+        ws[f'E{row}'] = entry['phone']
+        ws[f'F{row}'] = entry['weekend']
+        ws[f'G{row}'] = entry['weekday']
+        ws[f'H{row}'] = entry['homebound']
+        ws[f'I{row}'] = entry['note']
+
+        return ws
+
+class EverythingSheet(MetaSheet):
+
+    def __init__(self, wb, trainingdata):
```

Comment:
```suggestion
    def __init__(self, wb, training_data):
```

We use snake case pretty consistently throughout the ECC Python code -- I suggest that you change `trainingdata` to `training_data` throughout the code.